### DISCUSSION TITLE
Calling convention with multiple stack pointers

### DISCUSSION THREAD

**Discussion by v-p-b** [NationalSecurityAgency/ghidra#5630] [Q&A]
I have a kinky compiler that apparently uses different registers for keeping track of local variables and (stack-based) parameters passed to a function.

I'd expect something like this, but I don't think there is an attribute to specify a register that would point to the "callee stack":

```
<pentry>
  <addr register="gpr13"/>
</pentry>
```

How do I specify a calling convention so that Ghidra can keep track of the "callee stack" during decompilation?

**Edit:** The two registers actually point inside the same memory region (the "same stack"), close to each other. In essence this looks similar to x86's ESP-EBP.

(This is a cross-post from [reddit](https://www.reddit.com/r/ghidra/comments/150k90d/calling_convention_with_multiple_stacks/), hope it's OK)

**Comment by virtuald** [comment]
I don't think there's a way to do it, but I'd like this also. I filed a bug at https://github.com/NationalSecurityAgency/ghidra/issues/5831.
